FirebaseAuth/Sources/Swift/AuthProvider/PhoneAuthProvider.swift

```diff
@@ -107,16 +107,23 @@ import Foundation
                                 uiDelegate: AuthUIDelegate? = nil,
                                 multiFactorSession: MultiFactorSession? = nil) async throws
       -> String {
-      return try await withCheckedThrowingContinuation { continuation in
```

Comment:
We should only do the implementation in one of the async or callback variation of the API. Most of them are currently done in the callback one. We should eventually move to async, but then map the callback version to it. I've started this transition in #13246